catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloader.java

```diff
@@ -163,16 +163,15 @@ public ResourceResponse setupDownload(Metacard metacard,
         if (downloaderConfig.isCacheEnabled()) {
 
             CacheKey keyMaker = null;
-
+            String key = null;
             try {
                 keyMaker = new CacheKey(metacard, resourceResponse.getRequest());
+                key = keyMaker.generateKey();
```

Comment:
Agreed. I am returning from the catch block looking for feedback on it. But the response is already created and does not change in the method so it should be okay to still return.